app/controllers/reviews_controller.rb

```diff
@@ -12,6 +12,48 @@ def create
     end
   end
 
+  def edit
+    review = Review.find(params[:id])
+    if signed_in? && current_user == review.user
+      @review = Review.find(params[:id])
+      @tattoo = @review.tattoo
+    elsif !signed_in?
+      authenticate_user!
+    else
+      flash[:notice] = 'You have no permission to edit this posting'
+      redirect_to tattoo_path(tattoo)
+    end
+  end
+
+  def update
+    @review = Review.find(params[:id])
+    @tattoo = @review.tattoo
+    if @review.update_attributes(review_params)
+      flash[:notice] = 'Review successfully updated.'
+      redirect_to tattoo_path(@tattoo)
+    elsif !signed_in?
+      authenticate_user!
+    else
```

Comment:
I think you have these `if` clauses in the wrong order - FIRST you're trying to update the post, and then only if that does not work are you checking to see if the user is signed in!